src/streaming/protection/controllers/ProtectionController.js

```diff
@@ -75,14 +78,16 @@ function ProtectionController(config) {
         protDataSet,
         sessionType,
         robustnessLevel,
-        keySystem;
+        keySystem,
+        cmcdModel;
 
     function setup() {
         logger = debug.getLogger(instance);
         pendingNeedKeyData = [];
         mediaInfoArr = [];
         sessionType = 'temporary';
         robustnessLevel = '';
+        cmcdModel = CmcdModel(context).getInstance();
```

Comment:
We should prefer to dependency injection here:  An instance of CMCD model is instantiated in MediaPlayer.js . Add this to ProtectionController in `MediaPlayer.detectProtection()`